spec/rubocop/cop/lint/out_of_range_regexp_ref_spec.rb

```diff
@@ -133,4 +133,146 @@
       puts $2
     RUBY
   end
+
+  it 'does not register an offense when in range references are used inside a when clause' do
+    expect_no_offenses(<<~RUBY)
+      case "foobar"
+      when /(foo)(bar)/
+        $2
+      end
+    RUBY
+  end
+
+  it 'registers an offense when out of range references are used inside a when clause' do
+    expect_offense(<<~RUBY)
+      case "foobar"
+      when /(foo)(bar)/
+        $3
+        ^^ Do not use out of range reference for the Regexp.
+      end
+    RUBY
+  end
+
+  it 'uses the minimum number of captures for when clauses with multiple conditions' do
+    expect_offense(<<~RUBY)
+      case "foobarbaz"
+      when /(foo)(bar)/, /(bar)baz/
+        $2
+        ^^ Do not use out of range reference for the Regexp.
```

Comment:
No offense here, $2 might be `'bar'`, right?

Reply:
My reasoning was that the backref needs to be valid for _all_ branches, but now I think you're right that getting nil here in some cases could be an intentional idiom, and my interpretation is too strict.